src/components/date-picker/_date-picker.scss

```diff
@@ -120,6 +120,10 @@
     }
   }
 
+  .#{$prefix}--date-picker__icon:nth-last-child(4) {
```

Comment:
Hi @alisonjoseph wondering what this selector for. Trying to qSA for this selector with/without the label, but couldn’t find any. Any insights…? Thanks!

Reply:
@asudoh I added a comment to the file also, but this sets the correct position of the calendar icon when there is an error message below.

Reply:
Thanks a lot for clarifying @alisonjoseph and sorry for getting back to this late. I may be wrong setting up the HTML structure, but turning on/off `data-invalid` to show the error message seems to reveal that we need more work here:

![single](https://user-images.githubusercontent.com/1259051/41325473-79708718-6ef5-11e8-9796-7e64da772532.gif)
![range](https://user-images.githubusercontent.com/1259051/41325474-7acefe3c-6ef5-11e8-8389-bb061d881786.gif)

`:nth-last-child()` assumes particular DOM structure and it’d be good if we have any way to avoid it… Thanks!

Reply:
Correct, this only works when there IS a label and there is validation. In order to fix the styling for no label + validation we will have to either change the html (breaking change) or add javascript. Was thinking we could merge this in and create a new issue for the use-case in your example above.

Reply:
Also, would there be form validation if there is not a label? I would say that if it is being used in a form, a label is required

Reply:
@alisonjoseph @tw15egan Thank you for giving it more thoughts on this topic! Let explain more about above movie - I thought that the form validation stuff works by adding or removing `data-invalid` attribute depending on the form value is invalid or not. Here's what I found with such test:

* Single: Misaligned icon if `data-invalid` is *not* present (valid value)
* Range: Misaligned icon if `data-invalid` *is* present (invalid value)

Good to better understand how it's addressed in future and how it's addressed interim (or I'm totally missing something). Thanks!